File: solved.ac/class2/11651/11651.dart
import 'dart:io';

class Point {
  int x;
  int y;

  Point(this.x, this.y);
}

void main() {
  int n = int.parse(stdin.readLineSync()!);

  List<Point> points = [];

  for (int i = 0; i < n; i++) {
    List<int> input =
        stdin.readLineSync()!.split(' ').map((e) => int.parse(e)).toList();
    points.add(Point(input[0], input[1]));
  }

  points.sort((a, b) {
    if (a.y == b.y) {
      return a.x - b.x;
    } else {
      return a.y - b.y;
    }
  });

  for (final point in points) {
    print('${point.x} ${point.y}');
  }
}

File: solved.ac/class2/11651/11651.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

type Point struct {
	x int
	y int
}

var stdio = bufio.NewReadWriter(
	bufio.NewReader(os.Stdin),
	bufio.NewWriter(os.Stdout),
)

func main() {
	defer stdio.Flush()

	var n int;

	fmt.Fscanln(stdio, &n)

	var points [100000]Point;

	for index := 0; index < n; index++ {
		fmt.Fscanln(stdio, &points[index].x, &points[index].y)
	}
	
	sort.Slice(points[:n], func(i, j int) bool {
		if points[i].y == points[j].y {
			return points[i].x < points[j].x
		}
		return points[i].y < points[j].y
	})

	for i := 0; i < n; i++ {
		fmt.Fprintln(stdio, points[i].x, points[i].y)
	}
}